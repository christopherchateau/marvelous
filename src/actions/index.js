export const storeCharacter = (character) => ({
  type: 'STORE_CHARACTER',
  character
})

export const updateStorageDetails = (currentIndex, length) => ({
  });
