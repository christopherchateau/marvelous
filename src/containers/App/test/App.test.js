import React from "react";
import { shallow } from "enzyme";
import { storeCharacter } from "../../../actions";
import { App, mapDispatchToProps, mapStateToProps } from "../index";

describe("App", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App
        dispatchStoreCharacter={jest.fn()}
        storedCharacters={[]}
        showFavorites={false}
      />
    );
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapDispatchToProps", () => {
    it("should dispatch storeCharacter when dispatchStoreCharacter is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = storeCharacter({ name: "Spider-Man", id: 1 });

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.dispatchStoreCharacter({ name: "Spider-Man", id: 1 });
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

  describe("mapStateToProps", () => {
    it("should return an array of characters", () => {
      const mockState = { characters: [{ name: "Spider-Man", id: 1 }] };
      const expected = [{ name: "Spider-Man", id: 1 }];
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps.storedCharacters).toEqual(expected);
    });

    it("should return count of stored characters", () => {
      const mockState = { showFavorites: false };
      const expected = false;
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps.showFavorites).toEqual(expected);
    });
  });

  describe("generateRandomId", () => {
    it("should generate a random ID between 1010801 and 1011428", () => {
      const randomId = wrapper.instance().generateRandomId();
      expect(randomId).toBeGreaterThanOrEqual(1010801);
      expect(randomId).toBeLessThanOrEqual(1011428);
    });
  });

  describe("stopDuplicates", () => {
    it("should stop duplicate characters", () => {
      wrapper = shallow(
        <App
          dispatchStoreCharacter={jest.fn()}
          storedCharacters={[{ name: "Spider-Man", id: 1 }]}
          showFavorites={false}
        />
      );
      expect(wrapper.instance().stopDuplicates(1)).toBeTruthy;
      expect(wrapper.instance().stopDuplicates(2)).toBeFalsy;
    });
  });
});
