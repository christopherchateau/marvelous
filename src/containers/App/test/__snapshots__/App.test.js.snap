// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatchStoreCharacter={[MockFunction]}
    localStoreCharacter={[MockFunction]}
    location={
      Object {
        "pathname": "/",
      }
    }
    showFavorites={false}
    storedCharacters={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/characters"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/favorites"
        />,
        false,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/characters",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/characters"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/favorites"
          />,
          false,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/characters",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
